### DISCUSSION TITLE
How to change the New Window action?

### DISCUSSION THREAD

**Discussion by halostatue** [wezterm/wezterm#3878] [Q&A] [answered]
I’m exploring using wezterm and can’t quite figure out how to make one thing behave like iTerm2 does, which I prefer to the current behaviour. That is:

- When creating a new split or tab, use the currently discovered directory (current behaviour)
- When creating a new window, do not use the currently discovered directory, but potentially fall back to `$HOME`.

This seems like it should be configurable, but I can’t figure out exactly what I would end up configuring to force the cwd to `$HOME` for the `New Window` action.

**Comment by wez** [comment, marked as answer]
You can redefine the key assignment for new window to set the cwd explicitly:

```lua
config.keys = {
  { key = 'n', mods = 'CMD', action = wezterm.action.SpawnCommandInNewWindow { cwd=wezterm.home_dir } },
}
```

**Reply by BatmanAoD** [reply]
This would be helpful as an example in [the default_cwd page](https://wezfurlong.org/wezterm/config/lua/config/default_cwd.html); I saw the link to the `SpawnCommand` documentation, but it seemed complicated so I just went searching until I found this issue.

**Reply by mwoodson** [reply]
Following up late on this.

@wez if you wouldn't mind answering this.

`SpawnWindow` is what the default command seems to be to create a new window. This was gathered when I dumped it. To me, adding the `{ cwd=wezterm.home_dir }` to this option seems to make a lot of sense, to Spawn a new window, with the default home dir. However, this `cwd` parameter isn't valid on the SpawnWindow.

`SpawnCommandInNewWindow` is what is recommended here. But to me, SpawnCommandInNewWindow sounds like you are going to run a totally new command in a new window. This function seems to accept the `cwd` parameter. But the name of this is super confusing.

Could you shed any light on this?

(I am really really loving Wezterm, this just confuses me, and doesn't make sense to me...

**Reply by wez** [reply]
SpawnWindow is a limited version of SpawnCommandInNewWindow. The latter gives you more options. If you don't specify a command or a working directory, the same defaults as used by SpawWindow are used

**Reply by mwoodson** [reply]
thanks for the reply.
